executor/aggfuncs/func_group_concat.go

```diff
@@ -295,10 +316,13 @@ func (h *topNRows) tryToAdd(row sortRow) (truncated bool) {
 			h.rows[0].buffer.Truncate(h.rows[0].buffer.Len() - int(debt))
 		} else {
 			h.currSize -= uint64(h.rows[0].buffer.Len()) + h.sepSize
+			for _, dt := range h.rows[0].byItems {
+				truncatedDatumMemSize += GetDatumMemSize(dt)
```

Comment:
I think `truncatedDatumMemSize` should be `h.currSize` minus previous `h.currSize`. so that we can remove `oldMem` and `newMem` in `UpdatePartialResult` function, the size of `byItems` has added in `range e.byItems`